elife/openjdk8.sls

```diff
@@ -0,0 +1,37 @@
+# 14.04, 16.04 and 18.04 have a "openjdk-8-jre-headless" package
+# 14.04 requires a ppa
+
+# remove oracle java (if it exists)
+
+oracle-ppa-removal:
+    pkgrepo.absent:
+        # https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkgrepo.html#salt.states.pkgrepo.absent
+        - ppa: webupd8team/java # no idea if this is correct
+
+oracle-java8-installer-removal:
+    pkg.purged:
+        - name: oracle-java8-installer
+        - require:
+            - oracle-ppa-removal
+
+{% if salt['grains.get']('osrelease') == "14.04" %}
+
+openjdk8 ppa:
+    pkgrepo.managed:
+        - ppa: jonathonf/openjdk
```

Comment:
does this require a key? e.g. https://github.com/elifesciences/builder-base-formula/blob/01b601ead7d331abb4d0714e5d597bda38a123ed/elife/gcloud.sls#L4-L16

Reply:
[doesn't appear to:](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkgrepo.html#salt.states.pkgrepo.managed)

> APT-BASED SYSTEMS
> ppa
>On Ubuntu, you can take advantage of Personal Package Archives on Launchpad simply by specifying the user and archive name. The keyid will be queried from launchpad and everything else is set automatically.